internal/controller/driver_controller.go

```diff
@@ -907,6 +930,7 @@ func (r *driverReconcile) reconcileNodePluginDeamonSet() error {
 								},
 								VolumeMounts: []corev1.VolumeMount{
 									utils.PluginDirVolumeMount,
+									utils.If(r.logRotation(), utils.LogDirVolumeMount(r.driver.Spec.Log.Rotation.LogHostPath, daemonSet.Name), corev1.VolumeMount{}),
```

Comment:
You should not use `utils.If` and an empty volume mount definition in the volume list.
See a previous comment for the reasoning